compiler/rustc_type_ir/src/relate.rs

```diff
@@ -571,15 +568,20 @@ pub fn structurally_relate_consts<I: Interner, R: TypeRelation<I>>(
     mut a: I::Const,
     mut b: I::Const,
 ) -> RelateResult<I, I::Const> {
-    debug!("{}.structurally_relate_consts(a = {:?}, b = {:?})", relation.tag(), a, b);
+    debug!("{}.structurally_relate_consts(a = {:?}, b = {:?})", std::any::type_name::<R>(), a, b);
```

Comment:
I feel like these debug assertions are kinda pointless actually. Maybe we should just remove them.